### ISSUE TITLE
Cast problem

### ISSUE THREAD

**Issue by robert-johansson** [opennars/opennars#551]
The cast problem previously addressed seems to still exist:

```prolog
<<(*,$1,$2) --> opposite> <=> <(*,$2,$1) --> opposite>>.
<(&&,<(*,$1,#2) --> opposite>,<(*,#2,$3) --> opposite>) ==> <$1 <-> $3>>.
<(*,A1,B2) --> opposite>.
<(*,A1,C2) --> opposite>.
<B2 <-> C2>?
20000
```

Answer <B2 <-> C2>. %1.00;0.45%
 ERR ClassCastException: org.opennars.language.Similarity cannot be cast to org.opennars.language.Inheritance [org.opennars.inference.CompositionalRules.addVariableCandidates(CompositionalRules.java:794),
.....

**Comment by robert-johansson** [comment]
Even simpler to reproduce:

```prolog
<A1 <-> B1>. :|:
10
<A1 <-> C1>. :|:
10
```

**Comment by patham9** [comment]
Fixed in v3.0.3_otherFixes
This branch will finalize v3.0.3 for which some exceptions are left to fix.
